contracts/Masks/DEFYMasks.sol

```diff
@@ -0,0 +1,214 @@
+    function mintFungibleMask(
+        address account,
+        uint256 id,
+        uint256 amount,
+        bytes memory data
+    ) public onlyRole(MINTER_ROLE) {
+        // Require tokenIds range to be 0-999
+        require(id < 1000, "DEFYMask: Id is not for a fungible mask");
+
+        _mint(account, id, amount, data);
+    }
+
+    // Mint Batch Fungible Masks Only
+    function mintBatchFungibleMasks(
+        address to,
+        uint256[] memory ids,
+        uint256[] memory amounts,
+        bytes memory data
+    ) public onlyRole(MINTER_ROLE) {
+        // Require tokenIds range to be 0-999
+        for (uint256 i = 0; i < ids.length; i++) {
+            require(ids[i] < 1000, "DEFYMask: Id is not for a fungible mask");
+        }
+        _mintBatch(to, ids, amounts, data);
+    }
+
+    function mintBatchMultiUserFungibleMasks(
+        address[] calldata addresses,
+        uint256[] calldata ids,
+        uint256[] calldata amounts
+    ) public onlyRole(MINTER_ROLE) {
+        require(addresses.length <= 80, "DEFYMask: Mint batch max count is 80");
+        require(
+            addresses.length == ids.length &&
+                addresses.length == amounts.length,
+            "DEFYMask: All arrays must be the same length"
+        );
+
+        // Require tokenIds range to be 0-999
+        for (uint256 i = 0; i < ids.length; i++) {
+            require(ids[i] < 1000, "DEFYMask: Id is not for a fungible mask");
+        }
+
+        bytes memory zeroBytes;
+
+        for (uint256 i = 0; i < addresses.length; i++) {
+            _mint(addresses[i], ids[i], amounts[i], zeroBytes);
+        }
+    }
+
+    // Mint NFT Masks Only
+    function mintNFTMask(address account) public onlyRole(MINTER_ROLE) {
+        // Get new Id from counter
+        uint256 maskId = _nftMaskIdCounter.current() + NON_FUNGIBLE_BASE;
+        _nftMaskIdCounter.increment();
+
+        require(maskId >= 1000, "DEFYMask: Internal counter error");
+
+        bytes memory zeroBytes;
+
+        _mint(account, maskId, 1, zeroBytes);
+    }
+
+    // Burn Masks
+    function burnMask(
+        address from,
+        uint256 id,
+        uint256 amount
+    ) public onlyRole(BURNER_ROLE) {
+        _burn(from, id, amount);
+    }
+
+    // Burns Fungible Masks Only
+    function burnBatchMask(
+        address from,
+        uint256[] memory ids,
+        uint256[] memory amounts
+    ) public onlyRole(BURNER_ROLE) {
+        _burnBatch(from, ids, amounts);
+    }
+
+    function setTokenTradingDisabledForToken(uint256 tokenId, bool disabled)
+        external
+        onlyRole(DEFAULT_ADMIN_ROLE)
+    {
+        _tokenTradingDisabled[tokenId] = disabled;
+    }
+
+    function _beforeTokenTransfer(
+        address operator,
+        address from,
+        address to,
+        uint256[] memory ids,
+        uint256[] memory amounts,
+        bytes memory data
+    ) internal override whenNotPaused {
+        for (uint256 i = 0; i < ids.length; i++) {
```

Comment:
Can we also add in a "soulbound whitelist" similar to how it works in the DEFYSwag contract.  This gives us the option of pre-minting a bunch to a DEFY hot wallet and selling them on the black market, while keeping them soulbound for the buyer.  You should be able to just directly steal that logic